lib/internal/warn.js

```diff
@@ -0,0 +1,60 @@
+'use strict';
+
+// A process Warning is similar in structure to an Error except that they
+// are emitted by the process.on('warn') event rather than being thrown.
+
+const _message = Symbol('message');
```

Comment:
@benjamingr ... largely to high them behind read-only getters. It's an alternative to using Object.defineProperty that just seems cleaner for me. I'm not sure how much precedent there is for the pattern, however.